service/device/registry/test/src/test/resources/features/DeviceRegistry.feature

```diff
@@ -16,237 +16,209 @@ Feature: Device Registry CRUD tests
+        Kapua should return an error.
+
+        Given I expect the exception "KapuaIllegalArgumentException" with the text "An illegal value was provided for the argument deviceCreator.clientId"
+        When I try to create devices with invalid symbols "#+*&,?>/" in name
+        Then An exception was thrown
+
+    #This scenario must be separated from the above scenario because upper scenario cannot contain (::) characters in a row.
+    Scenario: Creating a single device with clientID that contains invalid character
+        Create a single device with clientID invalid special character ::.
+        Kapua should return an error.
+
+        Given I expect the exception "KapuaIllegalArgumentException" with the text "An illegal value was provided for the argument deviceCreator.clientId"
+        When I create a device with name "testDevice::"
+        Then An exception was thrown
+
+    Scenario: Creating two device with the same clientID
+        Try to create two devices with same clientID.
+        Kapua should return an error.
+
+        Given I expect the exception "KapuaDuplicateNameException" with the text "An entity with the same name testDevice already exists."
+        When I create a device with name "testDevice"
+        And I create a device with name "testDevice"
+        Then An exception was thrown
+
+    Scenario: All device parameters must match the device creator
+        Create a test device and check whether it was created correctly. All the device
+        parameters must match the device creator specifications.
+
+        Given A device named "testDevice"
+        Then The device matches the creator parameters
+
+    Scenario: Case sensitiveness of named device searches
+        Searching by client ID is case sensitive.
+
+        Given A device named "CaseSensitiveTestName"
+        Then Named device registry searches are case sensitive
+
+    Scenario: Find device by registry ID
+        It must be possible to find a device in the registry by its registry ID.
+
+        Given A device named "TestDevice"
+        When I search for a device with the remembered ID
+        Then The device matches the creator parameters
+
+    Scenario: Find device by client ID
+        It must be possible to find a device in the registry by its Client ID.
+
+        Given A device named "TestDevice"
+        Then I find the device based on clientID "TestDevice"
+
+    Scenario: Try to find a device with an invalid registry ID
+        Searching for a nonexistent device should not raise any exception.
+        Empty list should be returned.
+
+        When I search for a device with a random ID
+        Then There is no such device
+
+    Scenario: Try to find a device with an invalid client ID
+        Searching for a nonexistent device should not raise any exception.
+        Empty list should be returned.
+
+        When I search for a device with a random client ID
```

Comment:
Can you please change the name of this step to: "When I search for a device with a nonexisting client ID"